### ISSUE TITLE
Defusal Without Death

### ISSUE THREAD

**Issue by asinck** [allejo/UselessMine#15]
With respect to the ePlayerDieEvent [here, before pull request](https://github.com/allejo/UselessMine/blob/master/UselessMine.cpp#L269).

I think that there might be an issue here. This case checks the mines when a player dies; this includes defused mines. In the typical case, this is fine: a player defuses a mine, and the mine goes off at the owner's location, the owner dies, and everything's nice. The problem would be if the defused mine SW goes off but *doesn't* kill the owner (for instance, if the owner is moving quickly or holding SH). When the owner *does* die, this message will trigger at that point, giving a seemingly random message. Note also that this means that the player won't get a message when the mine is defused, but when the owner dies. 

A quick logic walkthrough:

Owner lays a mine, and hops on the tracks around the map.
Victim with BD runs over the mine, which triggers on the track.
The SW misses the owner, who proceeds to drop off the track and get shot. 
Note that the mine was flagged as "defused", so when the ePlayerDieEvent triggers, the defused section will activate. The killer ID is set to the mine defuser, and the messages are sent.

So in summary, if the owner evades the defusal the messages will still be sent. I don't know if the points will be misassigned.

**Comment by allejo** [comment]
You bring up a good point. I originally had this to handle that situation but I think the 60 second delay would cause the issue you're talking about.

https://github.com/allejo/UselessMine/blob/8a1e350c770561737ed5ed62e84ace8f88f5041a/UselessMine.cpp#L99-L102

I wonder if BZFS is reusing shot IDs too quickly and that's why it's throwing off the shot ID check. I wonder if using explicit shot IDs when firing the world weapon would be better than letting the server give us one.

**Comment by asinck** [comment]
Would it be possible to just do a simple check to see if the killer was the server or not? If the defused mine killed the owner, then killer == server. If the mine missed the owner, then killer != server.

**Comment by allejo** [comment]
If I'm not mistaken, it'd be `253` in this case, not `BZ_SERVER` (a weird thing in the API). But it's worth testing this theory
